File: rust/src/models/update_add_on_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::price_id::PriceId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct UpdateAddOnRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_instances_per_subscription: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub price_id: Option<PriceId>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub self_serviceable: Option<bool>,
}

impl UpdateAddOnRequest {
    pub fn new() -> Self {
        Self {
            description: None,
            max_instances_per_subscription: None,
            name: None,
            price_id: None,
            self_serviceable: None,
        }
    }
}

File: typescript/src/models/updateAddOnRequest.ts
// this file is @generated
import { type PriceId, PriceIdSerializer } from "./priceId";

export interface UpdateAddOnRequest {
  description?: string | null;

  maxInstancesPerSubscription?: number | null;

  name?: string | null;

  priceId?: PriceId | null;

  selfServiceable?: boolean | null;
}

export const UpdateAddOnRequestSerializer = {
  _fromJsonObject(object: any): UpdateAddOnRequest {
    return {
      description: object["description"],
      maxInstancesPerSubscription: object["max_instances_per_subscription"],
      name: object["name"],
      priceId:
        object["price_id"] != null
          ? PriceIdSerializer._fromJsonObject(object["price_id"])
          : undefined,
      selfServiceable: object["self_serviceable"],
    };
  },

  _toJsonObject(self: UpdateAddOnRequest): any {
    return {
      description: self.description,
      max_instances_per_subscription: self.maxInstancesPerSubscription,
      name: self.name,
      price_id:
        self.priceId != null ? PriceIdSerializer._toJsonObject(self.priceId) : undefined,
      self_serviceable: self.selfServiceable,
    };
  },
};
